curve25519-dalek/src/edwards.rs

```diff
@@ -1291,16 +1341,8 @@ impl Debug for EdwardsPoint {
 impl group::Group for EdwardsPoint {
     type Scalar = Scalar;
 
-    fn random(mut rng: impl RngCore) -> Self {
-        let mut repr = CompressedEdwardsY([0u8; 32]);
-        loop {
-            rng.fill_bytes(&mut repr.0);
-            if let Some(p) = repr.decompress() {
-                if !IsIdentity::is_identity(&p) {
-                    break p;
-                }
-            }
-        }
+    fn random(rng: impl RngCore) -> Self {
+        Self::random(rng)
```

Comment:
Note to self: had to check this doesn't cause infinite recursion. It doesn't.

Reply:
Yeah, it's a little confusing calling a same-named inherent method from a trait method, but Rust prefers the inherent method to recursing

Reply:
Added a comment in [18cd950](https://github.com/dalek-cryptography/curve25519-dalek/pull/759/commits/18cd95024da1e507ce06f6af7f76071f9174753f)